tests/components/twitch/test_config_flow.py

```diff
@@ -0,0 +1,362 @@
+from tests.components.twitch.conftest import (
+    CLIENT_ID,
+    SCOPES,
+    TITLE,
+    TWITCH_AUTHORIZE_URI,
+)
+from tests.typing import ClientSessionGenerator
+
+
+async def test_full_flow(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    current_request_with_host: None,
+    mock_setup_entry,
+    twitch: TwitchMock,
+) -> None:
+    """Check full flow."""
+    result = await hass.config_entries.flow.async_init(
+        "twitch", context={"source": SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+
+    assert result["url"] == (
+        f"{TWITCH_AUTHORIZE_URI}?response_type=code&client_id={CLIENT_ID}"
+        "&redirect_uri=https://example.com/auth/external/callback"
+        f"&state={state}&scope={'+'.join(SCOPES)}"
+    )
+
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "channel123"
+    assert "result" in result
+    assert "token" in result["result"].data
+    assert result["result"].data["token"]["access_token"] == "mock-access-token"
+    assert result["result"].data["token"]["refresh_token"] == "mock-refresh-token"
+    assert result["result"].unique_id == "123"
+    assert result["options"] == {CONF_CHANNELS: ["internetofthings", "homeassistant"]}
+
+
+async def test_already_configured(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    current_request_with_host: None,
+    config_entry: MockConfigEntry,
+    mock_setup_entry,
+    twitch: TwitchMock,
+) -> None:
+    """Check flow aborts when account already configured."""
+    await setup_integration(hass, config_entry)
+    result = await hass.config_entries.flow.async_init(
+        "twitch", context={"source": SOURCE_USER}
+    )
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+
+    assert result["url"] == (
+        f"{TWITCH_AUTHORIZE_URI}?response_type=code&client_id={CLIENT_ID}"
+        "&redirect_uri=https://example.com/auth/external/callback"
+        f"&state={state}&scope={'+'.join(SCOPES)}"
+    )
+
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
```

Comment:
I would suggest to create a function for this code part as it is duplicated 5 times